src/pages/Analytics/AnalyticsTable.js

```diff
@@ -1,17 +1,70 @@
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import i18n from '@dhis2/d2-i18n'
 import PropTypes from '@dhis2/prop-types'
 import TableActions from '../../components/table-actions'
 import DialogDelete from '../../components/dialog/dialog-delete'
-import { removeSettingsFromList } from './helper'
+import {
+    createTEIValues,
+    populateAnalyticItem,
+    populateWHOItem,
+    removeSettingsFromList,
+    updateList,
+    WHO_NUTRITION,
+} from './helper'
+import { validateObjectByProperty } from '../../utils/validators/validateObjectByProperty'
+import DialogAnalyticsTEI from './DialogAnalyticsTEI'
 
 const AnalyticsTable = ({ rows, disableAll, handleRows }) => {
     const [specificSettings, setSpecificSettings] = useState({})
     const [openDeleteDialog, setOpenDeleteDialog] = useState(false)
+    const [openEditDialog, setOpenEditDialog] = useState(false)
+    const [disableSave, setDisableSave] = useState(false)
+
+    useEffect(() => {
+        if (specificSettings.type === WHO_NUTRITION) {
+            validateObjectByProperty(
```

Comment:
This effect is nearly identical to code in `NewAnalyticSetting`. In the future when you have time you could refactor this to a separate function that could be used by both components.